cmd/cluster-agent/api/v1/language_detection.go

```diff
@@ -37,21 +39,22 @@ func postDetectedLanguages(w http.ResponseWriter, r *http.Request) {
 	// Unmarshal the request body into the protobuf message
 	err = proto.Unmarshal(body, requestData)
 	if err != nil {
-		languagedetection.ErrorResponses.Inc()
 		http.Error(w, "Failed to unmarshal request body", http.StatusBadRequest)
+		languagedetection.ErrorResponses.Inc()
+		return
 	}
 
 	lp, err := languagedetection.NewLanguagePatcher()
-
 	if err != nil {
-		languagedetection.ErrorResponses.Inc()
 		http.Error(w, "Failed to get k8s apiserver client", http.StatusInternalServerError)
+		languagedetection.ErrorResponses.Inc()
+		return
 	}
 
-	// Patch annotations to deployments
-	lp.PatchAllOwners(requestData)
-
-	// Respond to the request
+	// Answer before patching
 	languagedetection.OkResponses.Inc()
 	w.WriteHeader(http.StatusOK)
+
+	// Patch annotations to deployments
+	lp.PatchAllOwners(requestData)
```

Comment:
I think it would be better if this call was asynchronous because patching can be very slow and if I understand correctly we process requests one by one.

Reply:
- I agree that having an asynchronous call might result in better performance.
- Not sure if the handler blocks incoming requests when it is processing an active request.